src/shogun/base/base_types.h

```diff
@@ -167,6 +167,20 @@ namespace shogun
 
 	template <typename T>
 	inline constexpr bool is_auto_value_v = is_auto_value<T>::value;
+
+
+	template<typename classType>
+	std::string_view name_lookup()
```

Comment:
could you just this to a more meaningful name please?

Reply:
@vigsterkr do you know if there is another way to lookup the class name and get the name as a string?

Reply:
Not sure if this is of any help: https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/lib/any.h#L856

Reply:
the problem is that it isn't cross platform, i.e. doesn't work with msvc